addons/sale_timesheet/static/src/components/so_line_field/so_line_field.js

```diff
@@ -13,7 +13,7 @@ export class SoLineField extends Many2OneField {
             update(value, params);
             if ( // field is unset AND the old & new so_lines are different
                 !this.props.record.data.is_so_line_edited &&
-                this.props.value[0] != (value[0] && value[0].id)
+                this.props.record.data.so_line[0] != value[0]?.id
```

Comment:
In fact, we could do that: 
```suggestion
                this.value[0] != value[0]?.id
```

since `value` is defined in `Many2oneField` component.

https://github.com/odoo/odoo/blob/d5d7a8a657ce35dff377fd0c2bec05cd88cab96c/addons/web/static/src/views/fields/many2one/many2one_field.js#L197-L199

Reply:
Hello @xavierbol 
Thanks for suggestions. I have made changes. :+1: